insights/parsers/openstack_server_list_all.py

```diff
@@ -0,0 +1,64 @@
+"""
+openstack server list --all command - Command
+=============================================
+Parsing and extracting data from output of command ``/usr/bin/openstack server list --all``.
+--------------------------------------------------------------------------------------------
+"""
+
+from insights import CommandParser, parser, LegacyItemAccess
+from insights.parsers import ParseException, SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.openstack_server_list_all)
+class OpenstackServerListAll(LegacyItemAccess, CommandParser):
+    """
+    The output of command ``/usr/bin/openstack server list --all`` is in table format,
+    contains 'Variable_name' and 'Value' two columns.
+    This parser will parse the table and set each variable as an class
+    attribute. The unparsable lines are stored in the ``bad_lines`` property list.
+    Sample command output::
+
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+        | ID                                   | Name         | Status | Networks               | Image          | Flavor     |
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+        | 410b05bb-59b7-4b4c-88e9-975c811d68da | compute-0    | ACTIVE | ctlplane=192.168.24.20 | overcloud-full | compute    |
+        | f891e98b-4df6-4c90-9bf1-39cf8ac900b0 | compute-1    | ACTIVE | ctlplane=192.168.24.9  | overcloud-full | compute    |
+        | 3d62cd7e-41d2-43dd-a5bf-5935bc319fae | controller-0 | ACTIVE | ctlplane=192.168.24.10 | overcloud-full | controller |
+        +--------------------------------------+--------------+--------+------------------------+----------------+------------+
+    """
+
+    def parse_content(self, content):
+        """
+        Parse output content table of command ``/usr/bin/openstack server list --all``.
+        Set each variable as an class attribute.
+        """
+        data = []
```

Comment:
You might want to consider using a combination of the insights-core helper functions [get_active_lines](https://insights-core.readthedocs.io/en/latest/api_index.html#insights.parsers.get_active_lines) and [parse_delimited_table](https://insights-core.readthedocs.io/en/latest/api_index.html#insights.parsers.parse_delimited_table) to parse this table.  For instance you could do something like the following:
```python
# Removes the lines starting with "+----"
table = get_active_lines(content, comment_char='+')
# Parses header line and each row using the "|" separator
servers = parse_delimited_table(table, delim='|')
```
This much will give you a list of dicts that looks something like this:
```python
>>> servers
[{'': '',
  'ID': '410b05bb-59b7-4b4c-88e9-975c811d68da',
  'Name': 'compute-0',
  'Status': 'ACTIVE',
  'Networks': 'ctlplane=192.168.24.20',
  'Image': 'overcloud-full',
  'Flavor': 'compute'},
 ...
]
```
You'd still have to deal with the empty key because of the leading `|`, but it would be less complicated.